pages/_app.tsx

```diff
@@ -1,67 +1,47 @@
-import { memo } from 'react';
-import NextApp, { AppContext, AppProps } from 'next/app';
-import {
-  ToastContainer,
-  CookiesTooltip,
-  migrationAllowCookieToCrossDomainCookieClientSide,
-  migrationThemeCookiesToCrossDomainCookiesClientSide,
-  CookieThemeProvider,
-} from '@lidofinance/lido-ui';
-import { GlobalStyle } from 'shared/ui';
-import { ModalProvider } from 'features/walletModal';
-import { ProviderWeb3 } from 'reef-knot/web3-react';
-import dynamics from 'config/dynamics';
-import { backendRPC } from 'config';
-import Head from 'next/head';
-import { AppWagmiConfig } from 'features/wagmi';
+import { FC } from 'react';
+import NextApp, { AppProps } from 'next/app';
+
+import { EVMWidgetApp } from '@lidofinance/eth-next-widget-app-evm';
+import { migrationAllowCookieToCrossDomainCookieClientSide } from '@lidofinance/lido-ui';
 
-// Migrations old cookies to new cross domain cookies
-migrationThemeCookiesToCrossDomainCookiesClientSide();
+import { backendRPC, dynamics, getBackendRPCPath } from 'config';
+import { HeaderActions } from 'features/headerActions';
+import { Navigation } from 'features/navigation';
+import { NoSSRWrapper } from 'shared/ui/noSSRWrapper';
+import { GlobalStyle } from 'shared/ui';
 
 // Migrations old allow cookies to new cross domain cookies
 migrationAllowCookieToCrossDomainCookieClientSide(
   'LIDO_WIDGET__COOKIES_ALLOWED',
 );
 
-const App = memo((props: AppProps): JSX.Element => {
-  const { Component, pageProps } = props;
-
-  return <Component {...pageProps} />;
-});
-App.displayName = 'App';
-
-const MemoApp = memo(App);
-
-const AppWrapper = (props: AppProps): JSX.Element => (
-  <>
-    <Head>
-      <title>TRP UI | Lido</title>
-    </Head>
-
-    <CookieThemeProvider>
-      <AppWagmiConfig>
-        {/* @ts-expect-error need to patch web3-react */}
-        <ProviderWeb3
-          defaultChainId={dynamics.defaultChain}
-          supportedChainIds={dynamics.supportedChains}
-          rpc={backendRPC}
-          walletconnectProjectId={dynamics.walletconnectProjectId}
-        >
-          <ModalProvider>
-            <MemoApp {...props} />
-          </ModalProvider>
-        </ProviderWeb3>
-      </AppWagmiConfig>
-
+const AppWrapper: FC<AppProps> = (props) => (
+  <NoSSRWrapper>
```

Comment:
Why do we wrap the whole app into NoSSR?

Reply:
because several troubles places in widget and in warehouse (possibly reef knot related)

Reply:
Last time these errors were related to the connect button, but not the whole app. Can you double-check that the whole app isn't working, and we need to wrap it in NoSSRWrapper?

Reply:
It wasn't much, but some info still was rendered 

<img width="546" alt="image" src="https://github.com/lidofinance/trp-ui/assets/103929444/8b6180c3-54b6-422c-9a81-53fbc7bb52bf">